Swap Camera for Image Entity

prairiesnpr opened the pull request:
This swaps the Camera entity to an Image entity.

Thomas55555 commented:
- It seems like there is text missing for "Enable Image"

<img width="256" alt="Screenshot 2023-07-11 204529" src="https://github.com/Thomas55555/husqvarna_automower/assets/59625598/d4005513-315d-41d6-bbd6-3df9c2514d2b">

- Shouldn't we change the state of the image to `image_last_updated`. So that we have a timestamp there instead of, the mower's state?

Thomas55555 commented on custom_components/husqvarna_automower/entity.py:
```diff
@@ -3,11 +3,8 @@
-from homeassistant.helpers.entity import DeviceInfo, Entity
-from homeassistant.helpers.update_coordinator import (
-    CoordinatorEntity,
-    DataUpdateCoordinator,
```
Ah good, that was unnecessary

Thomas55555 commented:
I've removed the 'HusqvarnaAutomowerStateMixin` class. Think, thats not relevant anymore. Is there a need to change the tests?

Thomas55555 commented on custom_components/husqvarna_automower/translations/en.json:
```diff
@@ -1,242 +1,243 @@
+    "image": {
+      "mower_img": {
+        "name": "Map",
+        "state": {
+          "cleaning": "Mowing",
+          "docked": "Docked",
+          "error": "Error",
+          "idle": "Idle",
+          "off": "Off",
+          "on": "On",
+          "paused": "Paused",
+          "returning": "Returning to dock"
         }
```
Can be deleted ?!?

Thomas55555 commented on custom_components/husqvarna_automower/translations/de.json:
```diff
@@ -64,7 +64,7 @@
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Mähen"
```
Can be deleted?!?!

Thomas55555 commented on custom_components/husqvarna_automower/translations/et.json:
```diff
@@ -1,243 +1,243 @@
+    "image": {
+      "mower_img": {
+        "name": "Kaart",
+        "state": {
+          "cleaning": "Niidab",
+          "docked": "Dokitud",
+          "error": "Viga",
+          "idle": "[%key:common::state::idle%]",
+          "off": "[%key:common::state::off%]",
+          "on": "[%key:common::state::on%]",
+          "paused": "[%key:common::state::paused%]",
+          "returning": "Läheb laadimisjaama"
         }
```
Can be deleted?!?

Thomas55555 commented on custom_components/husqvarna_automower/translations/fr.json:
```diff
@@ -24,7 +24,7 @@
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Tonte"
```
Can be deleted!?!?

Thomas55555 commented on custom_components/husqvarna_automower/translations/no.json:
```diff
@@ -49,7 +49,7 @@
-        "camera": {
+        "image": {
             "mower": {
                 "state": {
                     "cleaning": "Klipper"
```
Can be deleted!?!?!

Thomas55555 commented on custom_components/husqvarna_automower/translations/sv.json:
```diff
@@ -1,204 +1,204 @@
+    "image": {
+      "mower_img": {
+        "name": "Karta",
+        "state": {
+          "cleaning": "Klipper",
+          "docked": "Dockad",
+          "error": "Fel",
+          "idle": "Inaktiv",
+          "off": "Från",
+          "on": "Till",
+          "paused": "Pausad",
+          "returning": "Återvänder till dockan"
         }
```
Can be deleted!??!

Thomas55555 commented:
Look good. It feels like the image entity is faster than the camera entity.....
I've added some comments to the translations. I think there can be deleted some stuff. Can you have a look at this?
Further more I removed the `HusqvarnaAutomowerStateMixin`
I've you're okay with that, I think we can merge this. I assume, you have tested migration from the old to the new version? I didn't do that.

prairiesnpr commented:
> Look good. It feels like the image entity is faster than the camera entity..... I've added some comments to the translations. I think there can be deleted some stuff. Can you have a look at this? Further more I removed the `HusqvarnaAutomowerStateMixin` I've you're okay with that, I think we can merge this. I assume, you have tested migration from the old to the new version? I didn't do that.

I'll look this evening, I think the image entity is on average faster, but it seems to drag when using large images since we lose the ability to resize the image on the backend and that now has to be done in the front end.  I left the resize code in, hopefully that will get added back in HA and we can implement it.

I'm not certain about the state translations, thinking those should be removed in #511?  Will check to be sure.

I did test the migration, no issues found.

Didn't see any immediate concerns with [4dc8996](https://github.com/Thomas55555/husqvarna_automower/pull/512/commits/4dc8996372dd8cd9c497731a42f0b5c2cd1cbb57) , but want to load it up in a codespace and just double check.

Thomas55555 commented:
That are the state translation for the image. So they are not related to the other PR.

prairiesnpr commented:
OK, this should be ready to merge now.

fredoche1810 commented:
Hello @prairiesnpr @Thomas55555 ,
Thans for your update.
Before, I could take a snapshot, save it as JPG, and add the JPG to a notification.
Now that the camera entity doesn't exist anymore, I can't take a snapshot.
How can I attach a dynamic image to my notification now ?
Thanks in advance,

prairiesnpr commented:
> Hello @prairiesnpr @Thomas55555 , Thans for your update. Before, I could take a snapshot, save it as JPG, and add the JPG to a notification. Now that the camera entity doesn't exist anymore, I can't take a snapshot. How can I attach a dynamic image to my notification now ? Thanks in advance,

I'm using the same automation as before, just updated the camera entity to the image entity.

This is what I use now.

```
action:
  - variables:
      ha_url_var: !secret ha_url
      quest_update: '{{ states("sensor.aeinarr_error_quest_desc") }}'
  - service: notify.all_ios_devices
    data_template:
      message: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} {{ quest_update }}'
      data:
        attachment:
          url: '{{ha_url_var}}{{ state_attr("image.aeinarr_map", "entity_picture") }}'
          content-type: jpg
          hide-thumbnail: false
  - service: notify.all_android_devices
    data_template:
      title: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} Quest Update!'      message: '{{ state_attr("vacuum.aeinarr", "friendly_name") }} {{ quest_update }}'
      data:
        image: '{{ha_url_var}}{{ state_attr("image.aeinarr_map", "entity_picture") }}'
```

fredoche1810 commented:
Hello @prairiesnpr ,
Thanks for sharing your code. 
How do you manage !secret ha_url ?  
I tested your code with the internal url of homeassistant and it works if I'm on the LAN  
I tested with the external url and it works when I'm outside the LAN. 
How to make it working as well if connected to the LAN than when I'm outside the LAN ? 
Thanks,

prairiesnpr commented:
In my setup, ha_url is just the external URL to my instance. I don't use the internal URL. That said, try DROPPING the URL, I don't think it's required.
